### PR TITLE
Flutter runner

### LINKED ISSUE #89: Rewrite Flutter runner in Dart
For kicks... https://github.com/flutter/flutter/tree/master/packages/flutter_tools/templates/app/windows.tmpl

### PR DESCRIPTION
**Pull request by ilopX**
On request: https://github.com/timsneath/win32/issues/89

## Pre-launch Checklist
- [X] Add winApi OOP wrapper, for create window more simple
- [X] Add working flutter loader
- [X] Add command line tool `flutter_runner --bundle c:\proj\data --dll c:\some\flutter_windows.dll`
- [X] Add advanced search for dll and bundle
- [x] Add flutter bundle to demonstrate flutter loader
- [x] Resolve the issue on the package structure https://github.com/timsneath/win32/discussions/292
###  Found errors 
- Child windows are not redrawn if the application is minimized and restored

Current state (***this code is working***)
```dart
void main() {
  final bundlePath = FlutterFinder.searchBundleFolder('bundle');
  final flutterDllPath = FlutterFinder.searchDllFile();

  final bundle = Bundle.fromSourceDir(bundlePath);
  final flutterApi = FlutterApi.load(flutterDllPath);

  FlutterWindow(bundle, flutterApi)
    ..text = 'Flutter dart runner'
    ..size = Size(400, 500)
    ..show();

  NativeApp.run();
}
```

## Class view
image

### CODE REVIEW

File: flutter_runner/lib/win_api_gui_wrapper/native_app.dart [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      final createdClassName = createInfo.lpszClass.toDartString();
+      final nativeWindowAddress = createInfo.lpCreateParams.cast<Int64>();
+      try {
+        if (createdClassName == kWindowClassName) {
+          if (_winCtorStack.containsKey(nativeWindowAddress.value)) {
+            final nativeWindow = _winCtorStack[nativeWindowAddress.value];
+            _windows[hWnd] = nativeWindow!;
+            nativeWindow.onCreate();
+            return 0;
+          }
```
Comment (ilopX) [inline flutter_runner/lib/win_api_gui_wrapper/native_app.dart:51]: Wrap this in a separate methods
```dart
NativeWindow.fromHWND();
NativeWindow.pushNewWindow(createInfo);
```

File: flutter_runner/lib/win_api_gui_wrapper/native_app.dart [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    free(pWindowClassName);
+  }
+}
+
+String __windowClassName = '';
+
+String get _windowClassName {
+  if (__windowClassName.isEmpty) {
+    __windowClassName = kWindowClassName;
+    _registryWinClass();
```
Comment (ilopX) [inline flutter_runner/lib/win_api_gui_wrapper/native_app.dart:103]: Add deregistration method.

**Comment by timsneath** [comment]
Oh, this is AWESOME! I'm on a vacation trip without a Windows machine right now, but I'm eager to see what you've done!

**Comment by alexmercerind** [comment]
This is an awesome attempt. 👏

**Comment by ilopX** [comment]
Will be moved to separate repositories.
Discussion here https://github.com/timsneath/win32/discussions/292

**Pull request closed without merging**